### PR TITLE
Replace ethers-solc with foundry-compilers

### PR DESCRIPTION
**Pull request by shawn-zil**
ethers-solc (deprecated) is replaced with foundry-compilers.

### CODE REVIEW

File: zilliqa/src/contracts/mod.rs
Comment (JamesHinshelwood) [inline zilliqa/src/contracts/mod.rs:157]: May as well update this version to the latest stable one while we're here?

File: .github/workflows/ci.yaml
Comment (JamesHinshelwood) [inline .github/workflows/ci.yaml:52]: Do we need to run this on our own runners? Is it because of the disk space issues?

We have a limited number of self-hosted runners and thus the number of PRs that can be tested at a time is limited. This change will half that number.

If its not a lot of effort, maybe can we try a `du -sh *` (etc.) on the github-hosted runners and work out why this change started causing them to run out of space?

File: zilliqa/src/contracts/mod.rs
Comment (JamesHinshelwood) [inline zilliqa/src/contracts/mod.rs:160]:
```suggestion
```

File: zilliqa/tests/it/main.rs
Comment (JamesHinshelwood) [inline zilliqa/tests/it/main.rs:1115]: Why?

File: zilliqa/tests/it/main.rs
Comment (JamesHinshelwood) [inline zilliqa/tests/it/main.rs:1151]: I don't think this is necessary - You are just converted a `Bytes` to a `Bytes` via JSON?
```suggestion
```

File: zilliqa/tests/it/main.rs
Comment (JamesHinshelwood) [inline zilliqa/tests/it/main.rs:1149]: Maybe this is clearer? Say no if you disagree :) 
```suggestion
    // Convert from the `alloy` representation of an ABI to the `ethers` representation, via JSON
```

**Review by JamesHinshelwood — APPROVED** [review]
Will approve, but if possible lets try to investigate the CI runner issue before merging?

**Comment by shawn-zil** [comment]
> Will approve, but if possible lets try to investigate the CI runner issue before merging?

Yes. Something weird is happening. I'll remove it from auto merge for now.

File: zilliqa/tests/it/main.rs
Comment (shawn-zil) [reply zilliqa/tests/it/main.rs:1151]: yes, cos they're different 'Bytes'.

File: zilliqa/tests/it/main.rs
Comment (shawn-zil) [reply zilliqa/tests/it/main.rs:1149]: Done.

**Comment by shawn-zil** [comment]
> Will approve, but if possible lets try to investigate the CI runner issue before merging?

@JamesHinshelwood 
Sorted out the issues. Tests pass.

**Review by JamesHinshelwood — APPROVED** [review]

**Pull request merged**
